### PR TITLE
Adds watched_group_id for observed groups into history logs

### LINKED ISSUE #567: Progress > history in observation mode: should be the log of the observed groups
## Overview
When observing a group/user, the progress tab should contain results of the observed group/user (not the current user).
Screenshot 2021-07-28 at 16 28 39

## Acceptance criteria
- [ ] Use the "watched_group_id" parameter of the [item log service](https://france-ioi.github.io/algorea-devdoc/api/#operation/itemActivityLogForItem) to list the log of the observed group
- [ ] Test with multiple groups that it works as expected.

### PR DESCRIPTION
**Pull request by Iloveall**
Fixes #567

### CODE REVIEW

File: src/app/modules/item/pages/item-log-view/item-log-view.component.ts [hunk 1]
```diff
@@ -26,6 +26,7 @@ export class ItemLogViewComponent implements OnChanges, OnDestroy {
 
   @Input() itemData?: ItemData;
   @Input() isWatchingGroup = false;
+  @Input() watchingGroupId?: string;
```
Comment (smadbe) [inline src/app/modules/item/pages/item-log-view/item-log-view.component.ts:29]: Get this information directly from the session service as we did in the other places. (it is not a pure UI component (like a picker or a button) anyway).

**Review by smadbe — CHANGES_REQUESTED** [review]

File: src/app/modules/item/pages/item-log-view/item-log-view.component.ts (hunk 1 above)
Comment (Iloveall) [reply src/app/modules/item/pages/item-log-view/item-log-view.component.ts]: Fixed

File: src/app/modules/item/pages/item-log-view/item-log-view.component.ts [hunk 2]
```diff
@@ -25,19 +26,20 @@ interface Data {
...
-  @Input() watchingGroupId?: string;
 
   private readonly refresh$ = new Subject<void>();
   private readonly item$ = new ReplaySubject<Item>(1);
-  readonly state$ = this.item$.pipe(
-    distinct(),
-    switchMap((item: Item) => this.getData$(item)),
+  readonly state$ = combineLatest([
+    this.item$.pipe(distinct()),
+    this.sessionService.session$,
```
Comment (smadbe) [inline src/app/modules/item/pages/item-log-view/item-log-view.component.ts:34]: I suppose you can use `this.sessionService.watchedGroup$` directly.

File: src/app/modules/item/pages/item-log-view/item-log-view.component.ts [hunk 3]
```diff
@@ -81,7 +83,7 @@ export class ItemLogViewComponent implements OnChanges, OnDestroy {
       {
         field: 'item.user',
         header: $localize`User`,
-        enabled: this.isWatchingGroup && [ 'Chapter', 'Task', 'Course' ].includes(type),
+        enabled: this.sessionService.isCurrentlyWatching && [ 'Chapter', 'Task', 'Course' ].includes(type),
```
Comment (smadbe) [inline src/app/modules/item/pages/item-log-view/item-log-view.component.ts:86]: If you use observable to compute your states, you have to use them everywhere, but that should not be too difficult to pass this function a boolean.

**Review by smadbe — CHANGES_REQUESTED** [review]

File: src/app/modules/item/pages/item-log-view/item-log-view.component.ts (hunk 2 above)
Comment (Iloveall) [reply src/app/modules/item/pages/item-log-view/item-log-view.component.ts]: Fixed

File: src/app/modules/item/pages/item-log-view/item-log-view.component.ts (hunk 3 above)
Comment (Iloveall) [reply src/app/modules/item/pages/item-log-view/item-log-view.component.ts]: Fixed

**Review by smadbe — APPROVED** [review]

**Pull request merged**